src/controllers/users.controller.ts

```diff
@@ -60,6 +60,18 @@ class UsersController {
       next(error);
     }
   };
+
+  public manageGroups = async (req: Request, res: Response, next: NextFunction) => {
+    try {
+      const data = req.body;
+      const userId = Number(req.params.id);
+      const groupData = await this.userService.manageGroups(userId, data);
```

Comment:
There should be a check for `NaN`, line 67 will return `NaN` if the passed userId is not a valid number, and Prisma will likely throw a confusing error.
```suggestion
      const userId = Number(req.params.id);
      if (isNaN(userId)) {
          return res.status(400).json({ message: 'Invalid user ID' });
      }
      const groupData = await this.userService.manageGroups(userId, data);
```